hipcheck/src/plugin/parser/download_manifest.rs

```diff
@@ -1,55 +1,16 @@
+use super::extract_data;
+use crate::plugin::parser::ParseKdlNode;
+use crate::string_newtype_parse_kdl_node;
 use crate::{error::Error, hc_error};
 use kdl::{KdlDocument, KdlNode, KdlValue};
 use std::{fmt::Display, str::FromStr};
 
 // NOTE: the implementation in this crate was largely derived from RFD #0004
 
-#[allow(unused)]
-// Helper trait to make it easier to parse KdlNodes into our own types
-trait ParseKdlNode
-where
-	Self: Sized,
-{
-	/// Return the name of the attribute used to identify the node pertaining to this struct
-	fn kdl_key() -> &'static str;
-
-	/// Attempt to convert a `kdl::KdlNode` into Self
-	fn parse_node(node: &KdlNode) -> Option<Self>;
-}
-
-#[allow(unused)]
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Url(pub String);
```

Comment:
Maybe we should take it a step further than a `String` and wrap a `url::Url` type?

Reply:
I was not sure if it was always going to be a `Url`, if it is then we can definitely swap that from `String` to `url::Url`

Reply:
Based on the RDF I think its safe to assume it will always be `Url`